addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/NPMContentAccessHandler.java

```diff
@@ -446,25 +443,23 @@ public void generateHttpMetadataHeaders( final Transfer transfer, final HttpServ
         }
 
         final HttpExchangeMetadata metadata = new StoreHttpExchangeMetadata( request, responseWithLastModified );
-        OutputStream out = null;
-        try
+
+        try (OutputStream out = metaTxfr.openOutputStream( TransferOperation.GENERATE, false ))
         {
-            out = metaTxfr.openOutputStream( TransferOperation.GENERATE, false );
-            out.write( mapper.writeValueAsBytes( metadata ) );
+            if ( out != null )
+            {
+                out.write( mapper.writeValueAsBytes( metadata ) );
+            }
         }
         catch ( final IOException e )
         {
             logger.error( "Failed to write metadata for HTTP exchange to: {}. Reason: {}", metaTxfr, e );
         }
-        finally
-        {
-            IOUtils.closeQuietly( out );
-        }
 
         // npm will generate .tgz and version json metadata files from the package json file target,
         // which will also need the HttpExchangeMetadata for npm header check.
 
-        List<Transfer> generated = getGeneratedTransfers( transfer );
+        List<Transfer> generated = generatedTransfers.getGeneratedTransfers( transfer );
```

Comment:
This will work fine, if generatedTransfers is a parameter on the `generateHttpMetadataHeaders()` method. If you're calling this from multiple locations, you can always make it nullable, then do a null check here for `generatedTransfers` itself.